pubtools/pulplib/_impl/client/client.py

```diff
@@ -448,6 +448,14 @@ def update_repository(self, repository):
 
         return out
 
+    def get_repo_lock(self, repo_id):
```

Comment:
I think this might be better off living on the Repository class, and also with no 'get' prefix, so the caller would write something like:

```python
with repo.lock():
    repo.do_some_stuff(...)
```

Whether it lives on Client or on Repository is somewhat arbitrary, either will technically work.  I feel that Repository is more consistent with the rest of the API which exists today.

Also, it forces the caller to verify the repo's existence first. That might seem a negative (as you're forced to do one extra arguably unnecessary request), but in fact in many places where we first tried to make use of a repo without explicitly verifying it exists first, users had trouble understand the error messages if they passed the ID of a nonexistent repo.

It would be good also if we don't document the object returned by lock() and don't make it public API.  i.e. we document that it can be used as a context manager, and that's it.  This avoids us getting locked into an implementation that we might not be able to support long-term.